tests/src/testcommon.hh

```diff
@@ -187,139 +186,88 @@ template <typename NonLinearOperator>
-  for (const auto c : std::views::iota(0u, fe2.size())) {
-    const auto fineKey                        = fe2.localCoefficients().localKey(c);
-    const auto nodalPositionInChildCoordinate = referenceElement.position(fineKey.subEntity(), fineKey.codim());
-    fe.calculateAt(resultRequirements, nodalPositionInChildCoordinate, result);
-    Eigen::Vector3d computedResult = result.getResult(ResultType::linearStress);
-    const auto nodeIndex           = localScalarView.index(localScalarView.tree().localIndex(c))[0];
-    stressVector[nodeIndex]        = toDune(computedResult);
-    for (const auto voigtIndex : std::views::iota(0u, static_cast<unsigned int>(gridDim * (gridDim + 1) / 2))) {
-      const auto FEStressComponent       = stressVector[nodeIndex][voigtIndex];
-      const auto expectedStressComponent = expectedStress[c][voigtIndex];
-      const bool isStressCorrect
-          = Dune::FloatCmp::eq<double, Dune::FloatCmp::CmpStyle::absolute>(FEStressComponent, expectedStressComponent);
-      t.check(isStressCorrect) << "Stress component " << voigtIndex << " at node " << c << " is not correctly computed"
-                               << messageIfFailed;
-    }
-  }
-  return t;
-}
-
-template <typename NonLinearOperator, typename FiniteElement>
-[[nodiscard]] auto checkLinearStressOfSimplex(NonLinearOperator& nonLinearOperator, FiniteElement& fe,
-                                              const std::string& messageIfFailed = "") {
+[[nodiscard]] auto checkLinearStress(NonLinearOperator& nonLinearOperator, FiniteElement& fe,
+                                     const std::string& messageIfFailed = "") {
   constexpr int dim = FiniteElement::Traits::mydim;
 
   static_assert(std::remove_cvref_t<decltype(fe.localView().tree())>::degree() == dim,
-                "The test to check linear stress is only supported with the powerBasis being two-dimensional");
-  assert(fe.localView().element().type() == Dune::GeometryTypes::simplex(dim)
-         && "The test to check linear stress is only supported for simplices");
+                "The test to check linear stress is only supported with the powerBasis having the same power as the "
+                "element dimension");
+  assert(fe.localView().element().type() == Dune::GeometryTypes::cube(dim)
+         or fe.localView().element().type() == Dune::GeometryTypes::simplex(dim)
+                && "The test to check linear stress is only supported for cubes");
   static_assert(
       std::is_same_v<
           std::remove_cvref_t<decltype(fe.localView().tree().child(0))>,
           Dune::Functions::LagrangeNode<std::remove_cvref_t<decltype(fe.localView().globalBasis().gridView())>, 1>>,
       "The test to check linear stress is only supported for a linear Lagrange basis");
 
-  Dune::TestSuite t("Linear Stress check of" + std::to_string(dim) + "D  simplex element");
+  Dune::TestSuite t("Linear Stress check of" + std::to_string(dim) + "D  cube element");
```

Comment:
This is also out of sync?

Reply:
You can get the name via https://gitlab.dune-project.org/core/dune-geometry/-/blame/master/dune/geometry/type.hh?ref_type=heads#L421